packages/wrangler/src/dev.tsx

```diff
@@ -472,6 +476,54 @@ export async function startDev(args: StartDevOptions) {
 		const devEnv = new DevEnv();
 
 		if (args.experimentalDevEnv) {
+			const teardownRegistryPromise = devRegistry(async (registry) => {
+				const boundWorkers = await getBoundRegisteredWorkers(
+					{
+						name: devEnv.config.latestConfig?.name,
+						services: extractBindingsOfType(
+							"service",
+							devEnv.config.latestConfig?.bindings
+						),
+						durableObjects: {
+							bindings: extractBindingsOfType(
+								"durable_object_namespace",
+								devEnv.config.latestConfig?.bindings
+							),
+						},
+					},
+					registry
+				);
+
+				// Make sure we're not patching an empty config
+				if (!devEnv.config.latestConfig) {
+					await events.once(devEnv, "configUpdate");
+				}
+
+				// Ignore the dev registry in remote mode
+				if (devEnv.config.latestConfig?.dev?.remote) {
+					return;
+				}
```

Comment:
What if we change from remote to local mode?

Reply:
removed